server/src/test/Parsing.test.ts

```diff
@@ -46,37 +46,48 @@ describe('Parsing Tests', () => {
         });
     });
 
-    describe('applyCollection + __where rule', () => {
+    describe('applyCollection', () => {
         it('should parse correctly an `applyCollection` instruction with __where keyword', (done) => {
             checkInputValidityForRule(
                 (parser) => parser.applyCollection(),
                 `applyCollection(LibCube:Measure, __where currentElement.includeInJointureConditions == true)`,
             );
             done();
         });
+
+        it('should parse correctly an `applyCollection` instruction with __operation keyword', (done) => {
+            checkInputValidityForRule(
+                (parser) => parser.applyCollection(),
+                `applyCollection(toBeProcessedfactsSelections , __operation LibCube:JointureFactsSelection::jointure)`,
+            );
+            done();
+        });
     });
 
-    describe('applyCollection + __where as a function caller', () => {
+    describe('chainedCall rule', () => {
         it('should parse correctly an `applyCollection` instruction with __where keyword used as a function caller', (done) => {
             checkInputValidityForRule(
                 (parser) => parser.chainedCall(),
                 `applyCollection(LibCube:Measure, __where currentElement.includeInJointureConditions == true).toList()`,
             );
             done();
         });
-    });
 
-    describe('applyCollection + __operation rule', () => {
-        it('should parse correctly an `applyCollection` instruction with __operation keyword', (done) => {
+        it('should parse correctly a chained call with indexed accesses', (done) => {
```

Comment:
I see only one call here, judging by the number of parentheses pairs. Is that normal? How could it be “chained”, then? :thinking:

Reply:
kay